src/compiler/crystal/semantic/type_lookup.cr

```diff
@@ -249,7 +249,7 @@ class Crystal::Type
         end
 
         # Check the case of T resolving to a number
-        if type_var.is_a?(Path) && type_var.names.size == 1
+        if type_var.is_a?(Path)
```

Comment:
@asterite I'm curious, could you explain why removing this fixes the issue?

Reply:
Previously it works with paths of size one (`Foo`, `Bar`, etc.), now it works with any size (`Foo::Bar`, `Foo::Bar::Baz`, etc.). The size is the number of types in the path. I don't know why I originally restricted it to one.